### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ben-bay**
A starting point for the CONTRIBUTING file.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Merlin is open source software, and we welcome contribution!
+
+Our default branch is `master`. That's the stable branch, for users.
+
+`develop` is the branch you want to fork from and make pull requests into.
+
+Before issuing a pull request, please [squash your commits](https://github.com/LLNL/merlin/wiki/Squash-commits) down to one concise commit message.
```
Comment (lucpeterson) [inline CONTRIBUTING.md]: include a link to here: https://merlin.readthedocs.io/en/latest/merlin_developer.html

Comment (ben-bay) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Merlin is open source, and we welcome contribution!
+
+Our default branch is `develop`. That's the branch you want to fork from and make pull requests into.
```
[the remark this answers is not in the document]
Comment (lucpeterson) [reply CONTRIBUTING.md]: Change to:
We follow a gitflow - like model (http://www.dalescott.net/using-gitflow-with-githubs-fork-pull-model/). To contribute, fork your own copy. All changes are done via pull requests from your fork into the appropriate main repository branch. Features and additions into develop, hot fixes into master

[the remark this answers is not in the document]
Comment (ben-bay) [reply CONTRIBUTING.md]: @lucpeterson changed.

**Pull request merged**